osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/MonitoringService.java

```diff
@@ -26,33 +26,37 @@
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetPowerQualityProfileRequestDataDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.PeriodicMeterReadsRequestDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.ReadAlarmRegisterRequestDto;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service(value = "dlmsDeviceMonitoringService")
 public class MonitoringService {
 
-    @Autowired
     private GetPeriodicMeterReadsCommandExecutor getPeriodicMeterReadsCommandExecutor;
-
-    @Autowired
     private GetPeriodicMeterReadsGasCommandExecutor getPeriodicMeterReadsGasCommandExecutor;
-
-    @Autowired
     private GetActualMeterReadsCommandExecutor actualMeterReadsCommandExecutor;
-
-    @Autowired
     private GetActualMeterReadsGasCommandExecutor actualMeterReadsGasCommandExecutor;
-
-    @Autowired
     private ReadAlarmRegisterCommandExecutor readAlarmRegisterCommandExecutor;
-
-    @Autowired
     private GetPowerQualityProfileCommandExecutor getPowerQualityProfileCommandExecutor;
-
-    @Autowired
     private ClearAlarmRegisterCommandExecutor clearAlarmRegisterCommandExecutor;
 
+    public MonitoringService(GetPeriodicMeterReadsCommandExecutor getPeriodicMeterReadsCommandExecutor,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 7 parameters, which is greater than 6 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00107)